mocpy/moc.py

```diff
@@ -2,427 +2,360 @@
-        except ImportError:
-            from urllib.parse import urlencode
+        Load a moc from a set of fits images
 
-        return cls.from_url('%s?%s' % (MOC.MOC_SERVER_ROOT_URL, urlencode({'ivorn': ivorn, 'get': 'moc', 'order': int(np.log2(nside))})))
+        Parameters
+        ----------
+        path_l : [str]
+            the path list where the fits image are located
+        moc_order : int
+            moc resolution
 
-    @classmethod        
-    def from_url(cls, url):
-        from astropy.utils.data import download_file
+        Returns
+        -------
+        moc : `~mocpy.moc.MOC`
+            the union of all the moc from path_l
 
-        path = download_file(url, show_progress=False, timeout=60)
-        return cls.from_file(path)
-        
-    def add_uniq_ipix(self, uniq_ipix):
-        """
-        add a UNIQ pixel to the current MOC
         """
-        order, ipix = utils.uniq2orderipix(uniq_ipix)
-        self.add_pix(order, ipix)
-       
-    def add_pix_list(self, order, ipix_list, nest=True):
-        """
-        add a list of HEALPix pixel number at a given order
-        to the current object
-        """
-        for ipix in ipix_list:
-            self.add_pix(order, ipix, nest)
 
-    def _ensure_consistency(self):
-        """
-        Force IntervalSet internal consistency
-        """
-        self._interval_set.intervals
+        from astropy.io import fits
+        moc = MOC()
+        for path in path_l:
+            header = fits.getheader(path)
+            current_moc = MOC.from_image(header=header, moc_order=moc_order)
+            moc = moc.union(current_moc)
+
+        return moc
 
-    def add_pix(self, order, ipix, nest=True):
+    @classmethod
+    def from_vizier_table(cls, table_id, nside=256):
         """
-        add a given HEALPix pixel number to the current object
+        Create a `~mocpy.moc.MOC` object from a VizieR table
+
+        Parameters
+        ----------
+        table_id : int
+            table index
+        nside : int, optional
+            256 by default
+
+        Returns
+        -------
+        result : `~mocpy.moc.MOC`
+            the created moc
+
         """
-        self._counter += 1
-        # force consistency to prevent too large interval array
-        if self._counter == 1000:
-            self._ensure_consistency()
-            self._counter = 0
 
-        if order > MOC.HPY_MAX_NORDER:
-            raise Exception('norder can not be greater than MOC max norder')
-        
-        if not nest:
-            ipix = ring2nest(1 << order, ipix)
+        nside_possible_values = (8, 16, 32, 64, 128, 256, 512)
+        if nside not in nside_possible_values:
+            raise ValueError('Bad value for nside. Must be in {0}'.format(nside_possible_values))
 
-        p1 = ipix
-        p2 = ipix + 1
-        shift = 2 * (MOC.HPY_MAX_NORDER - order)
+        result = cls.from_ivorn('ivo://CDS/' + table_id, nside)
```

Comment:
Docstring says `table_id` is int, and here you add to a string. I don't think that can work!?

Reply:
Good eyes, table_id is a string. thanks !